src/core/server/root/index.ts

```diff
@@ -99,4 +91,43 @@ export class Root {
     await this.server.stop();
     this.server = undefined;
   }
+
+  private async setupLogging() {
+    function onLogUpgradeError(err: Error) {
+      // This specifically console.logs because we were not able to configure the logger.
+      // tslint:disable no-console
+      console.error('Configuring logger failed:', err);
+    }
+
+    // We should try to upgrade logger as soon as the root starts and shut root
+    // down if we can't do that, and subscribe to the consequent updates only if
+    // initial upgrade succeeded.
+    const config$ = this.configService.atPath('logging', LoggingConfig);
```

Comment:
__note:__ It would be great if we can define the reference approach for things like that: block on first emission and then just listen. Or maybe I'm looking at it from the wrong angle... Please, let me know what you think.

Reply:
The problem with `shareReplay(1)` is that it doesn't unsubscribe from the parent stream when the child stream unsubscribes, so if the logging service was stopped the subscription to the config service would stay open until the config service completed the `this.configService.atPath('logging', LoggingConfig)` stream, so it would be open forever.